src/main/java/com/structures/helpers/Structure.java

```diff
@@ -0,0 +1,422 @@
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.VertexBuffer;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityList;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumBlockRenderType;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.Mirror;
+import net.minecraft.util.Rotation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import net.minecraft.world.gen.structure.template.PlacementSettings;
+import net.minecraft.world.gen.structure.template.Template;
+import net.minecraftforge.client.ForgeHooksClient;
+import net.minecraftforge.client.MinecraftForgeClient;
+import net.minecraftforge.client.model.pipeline.LightUtil;
+
+/**
+ * Structure class, used to store, create, get structures.
+ */
+public class Structure
+{
+    /**
+     * Rotation by 90 degree.
+     */
+    private static final double  NINETY_DEGREE = 90D;
+    /**
+     * Template of the structure.
+     */
+    private Template          template;
+    private Minecraft         mc;
+    private PlacementSettings settings;
+
+    private static final int   COLOR_A    = 255;
+    private static final int   COLOR_R    = 255;
+    private static final int   COLOR_G    = 255;
+    private static final int   COLOR_B    = 255;
+
+    /**
+     * Constuctor of Structure, tries to create a new structure.
+     * @param world with world.
+     * @param structureName name of the structure (at stored location).
+     * @param settings it's settings.
+     */
+    public Structure(@Nullable World world, String structureName, PlacementSettings settings)
+    {
+        InputStream inputstream = MinecraftServer.class.getResourceAsStream("/assets/" + Constants.MOD_ID + "/schematics/" + structureName + ".nbt");
+
+        if(world == null || world.isRemote)
+        {
+            this.settings = settings;
+            this.mc = Minecraft.getMinecraft();
+        }
+        if (inputstream == null)
+        {
+            return;
+        }
+
+        try
+        {
+            this.template = readTemplateFromStream(inputstream);
+        }
+        catch (IOException e)
+        {
+            Log.getLogger().warn(String.format("Failed to load template %s", structureName), e);
+        }
+        finally
+        {
+            IOUtils.closeQuietly(inputstream);
+        }
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)